src/aind_metadata_mapper/fib/session.py

```diff
@@ -0,0 +1,295 @@
+from aind_metadata_mapper.fib.models import JobSettings
+from aind_metadata_mapper.fib.utils import (
+    extract_session_start_time_from_files,
+    extract_session_end_time_from_files,
+    verify_output_file,
+)
+
+
+class ETL(GenericEtl[JobSettings]):
+    """Creates fiber photometry session metadata using an ETL pattern.
+
+    This class handles the full lifecycle of session metadata creation:
+    - Extracting timing information from data files
+    - Transforming raw data into standardized session objects
+    - Loading/saving session metadata in a standard format
+
+    The ETL process ensures that all required metadata fields are populated
+    and validates the output against the AIND data schema.
+    """
+
+    def __init__(self, job_settings: Union[str, JobSettings]):
+        """Initialize ETL with job settings.
+
+        Args:
+            job_settings: Either a JobSettings object or a JSON string that can
+                be parsed into one. The settings define all required parameters
+                for the session metadata, including experimenter info, subject
+                ID, data paths, etc.
+
+        Raises:
+            ValidationError: If the provided settings fail schema validation
+            JSONDecodeError: If job_settings is a string but not valid JSON
+        """
+        if isinstance(job_settings, str):
+            job_settings = JobSettings(**json.loads(job_settings))
+        super().__init__(job_settings)
+
+    def _extract(self) -> JobSettings:
+        """Extract metadata from job settings and data files.
+
+        This method attempts to extract session start and end times from the
+        data files if they are not explicitly provided in the job settings.
+        It looks for timestamps in file names and CSV data to determine the
+        temporal bounds of the session.
+
+        Returns:
+            JobSettings: Updated settings object with extracted timing info
+
+        Notes:
+            If data_directory is not provided or no valid timestamps are found,
+            the original settings are returned unchanged.
+        """
+        settings = self.job_settings
+        logging.info("Starting metadata extraction")
+
+        if (
+            not hasattr(settings, "session_start_time")
+            or settings.session_start_time is None
+        ):
+            if hasattr(settings, "data_directory") and settings.data_directory:
+                logging.info(
+                    "Extracting session start time from "
+                    f"{settings.data_directory}"
+                )
+                return update_job_settings_with_times(
+                    settings, settings.data_directory
+                )
+            else:
+                logging.warning(
+                    "No data_directory provided and "
+                    "no session_start_time specified"
+                )
+
+        return settings
+
+    def _create_stream(
```

Comment:
This method could be decorated as a `@staticmethod` and the `self` arg can be removed. Alternatively, the settings could potentially be accessed directly via `self.job_settings`.

Reply:
Removed entirely.